lib/elixir_sense/providers/suggestion/reducers/struct.ex

```diff
@@ -0,0 +1,215 @@
+defmodule ElixirSense.Providers.Suggestion.Reducers.Struct do
+  @moduledoc false
+
+  alias ElixirSense.Core.Introspection
+  alias ElixirSense.Core.Metadata
+  alias ElixirSense.Core.Source
+  alias ElixirSense.Core.State
+  alias ElixirSense.Core.Struct
+
+  @type field :: %{
+          type: :field,
+          subtype: :struct_field | :map_key,
+          name: String.t(),
+          origin: String.t() | nil,
+          call?: boolean
+        }
+
+  @doc """
+  A reducer that adds suggestions of struct fields.
+  """
+  def add_fields(hint, text_before, env, buffer_metadata, acc) do
+    case find_struct_fields(hint, text_before, env, buffer_metadata) do
+      {[], _} ->
+        {:cont, acc}
+
+      {fields, nil} ->
+        {:halt, %{acc | result: fields}}
+
+      {fields, :maybe_struct_update} ->
+        {:cont, %{acc | result: fields, reducers: [:populate_common, :variables]}}
```

Comment:
IMO attributes and functions should be included here

Reply:
Yeah, good catch. Although is more common to see the update syntax being used with variables, functions and attributes are valid suggestions. I'll update it.